src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -172,4 +176,16 @@ public String getOptionalArrayTypeName() {
   public String getDynamicReturnTypeName(Method method, MethodConfig methodConfig) {
     return getNotImplementedString("SurfaceNamer.getDynamicReturnTypeName");
   }
+
+  public String getApiSettingsClassName(Interface service) {
+    return className(Name.upperCamel(service.getSimpleName(), "Settings"));
+  }
+
+  public String getTestClassName(Interface service) {
+    return className(Name.upperCamel(service.getSimpleName(), "Test"));
+  }
+
+  public String getMockServiceClassName(Interface service) {
+    return className(Name.upperCamel("Mock", service.getSimpleName()));
+  }
```

Comment:
The issue is when you want to override one of these strings for a specific language; it would explode the complexity even more if this was spread over the code.